django_social_share/templatetags/social_share.py

```diff
@@ -171,3 +171,41 @@ def post_to_reddit(context, title, obj_or_url=None, link_text='Post to Reddit'):
     context = post_to_reddit_url(context, title, obj_or_url)
     context['link_text'] = link_text
     return context
+
+
+def _compose_zap(text, url=None):
+    MESSAGE_MAX_NUMBER_OF_CHARACTERS = 500
+    MESSAGE_LINK_LENGTH = 150
+
+    url_length = len(' ') + MESSAGE_LINK_LENGTH if url else 0
+    total_length = len(text) + url_length
+
+    if total_length > MESSAGE_MAX_NUMBER_OF_CHARACTERS:
+        text = text[:(MESSAGE_MAX_NUMBER_OF_CHARACTERS - url_length - 1)] + "…"  # len("…") == 1
```

Comment:
You can use [`django.utils.text.Truncator`](https://github.com/django/django/blob/master/django/utils/text.py#L56) for truncating the text